File: GestionEvenementBack/src/main/java/tn/eesprit/gestionevenementback/Entities/Ordre.java
package tn.eesprit.gestionevenementback.Entities;
import javax.persistence.*;

import lombok.*;
import lombok.experimental.FieldDefaults;
import java.io.Serializable;
import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class Ordre implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer orderId;
    Integer quantity;
    Date dateLimitCommande;
    @ManyToOne
    Logistique logistique;
    @ManyToOne
    Product product;
    @ManyToOne
    Transport transport;
    @ManyToOne
    Housing housing;


}

File: GestionEvenementFront/src/app/_Models/Ordre.ts
import {Product} from "./Product";
import {Logistique} from "./Logistique";
import {Transport} from "./Transport";
import {Housing} from "./Housing";

export class Ordre {
  orderId:string ;
  dateLimitCommande:Date;
  quantity:number;
  initialQuantity:number;
  logistique:Logistique;
  product:Product ;
  transport:Transport ;
  housing:Housing ;

}
